### PR TITLE
update ci status

### PR DESCRIPTION
**Pull request by ebattat**
Fixes:

1. Add odf disk count to ci status
2. Add latest benchmark runner commit id to ci status:

e.g. 
`https://github.com/redhat-performance/benchmark-runner/commit/97e5724fcb01f81068e86fd26e7ef4fde56f269c`

### CODE REVIEW

File: .github/workflows/Nightly_Func_Env_CI.yml [hunk 1]
```diff
@@ -278,16 +278,21 @@ jobs:
...
         # get repository last id
-        declare -a repositories=('benchmark-operator' 'benchmark-wrapper')
+        declare -a repositories=('redhat-performance/benchmark-runner' 'cloud-bulldozer/benchmark-operator' 'cloud-bulldozer/benchmark-wrapper')
         for repository in "${repositories[@]}"
         do
-            git clone "https://github.com/cloud-bulldozer/$repository" "$RUNNER_PATH/$repository"
+            git clone "https://github.com/$repository" "$RUNNER_PATH/$repository"
             pushd "$RUNNER_PATH/$repository"
-            if [[ $repository == 'benchmark-operator' ]]
+            if [[ $repository == 'redhat-performance/benchmark-runner' ]]
```
Comment (RobertKrawitz) [inline .github/workflows/Nightly_Func_Env_CI.yml:286]: Do this as
```
case "$repository" in
    redhat-performance/benchmark-runner)
        ...
        ;;
    cloud-bulldozer/benchmark-operator
        ..
        ;;
    ...
    *)
        echo "Unexpected repository $repository"
        ;;
esac
```

File: .github/workflows/Nightly_Func_Env_CI.yml [hunk 2]
```diff
@@ -300,7 +305,7 @@ jobs:
...
+        podman run --rm -e KUBEADMIN_PASSWORD="$KUBEADMIN_PASSWORD" -e PIN_NODE_BENCHMARK_OPERATOR="$PIN_NODE_BENCHMARK_OPERATOR" -e PIN_NODE1="$PIN_NODE1" -e PIN_NODE2="$PIN_NODE2" -e ELASTICSEARCH="$ELASTICSEARCH" -e ELASTICSEARCH_PORT="$ELASTICSEARCH_PORT" -e ELASTICSEARCH_USER="$ELASTICSEARCH_USER" -e ELASTICSEARCH_PASSWORD="$ELASTICSEARCH_PASSWORD" -e BUILD_VERSION="$build_version" -e CI_STATUS="$status" -e CI_MINUTES_TIME="$ci_minutes_time" -e BENCHMARK_RUNNER_ID=$BENCHMARK_RUNNER_ID -e BENCHMARK_OPERATOR_ID="$BENCHMARK_OPERATOR_ID" -e BENCHMARK_WRAPPER_ID="$BENCHMARK_WRAPPER_ID" -e RUN_TYPE="func_ci" -e TIMEOUT="2000" -e log_level="INFO" -v "$KUBECONFIG_PATH":"$CONTAINER_KUBECONFIG_PATH" --privileged "quay.io/ebattat/benchmark-runner:latest"
```
Comment (RobertKrawitz) [inline .github/workflows/Nightly_Func_Env_CI.yml:308]:
```
-e BENCHMARK_RUNNER_ID="$BENCHMARK_RUNNER_ID"
```

File: .github/workflows/Nightly_Perf_Env_CI.yml [hunk 3]
```diff
@@ -269,16 +269,21 @@ jobs:
         CONTAINER_KUBECONFIG_PATH: ${{ secrets.CONTAINER_KUBECONFIG_PATH }}
       run: |
         # get repository last id
-        declare -a repositories=('benchmark-operator' 'benchmark-wrapper')
+        declare -a repositories=('redhat-performance/benchmark-runner' 'cloud-bulldozer/benchmark-operator' 'cloud-bulldozer/benchmark-wrapper')
         for repository in "${repositories[@]}"
         do
-            git clone "https://github.com/cloud-bulldozer/$repository" "$RUNNER_PATH/$repository"
+            git clone "https://github.com/$repository" "$RUNNER_PATH/$repository"
```
Comment (RobertKrawitz) [inline .github/workflows/Nightly_Perf_Env_CI.yml:275]: Same case...esac as above.

File: .github/workflows/Nightly_Perf_Env_CI.yml [hunk 4]
```diff
@@ -291,7 +296,7 @@ jobs:
...
+        podman run --rm -e KUBEADMIN_PASSWORD="$KUBEADMIN_PASSWORD" -e PIN_NODE_BENCHMARK_OPERATOR="$PIN_NODE_BENCHMARK_OPERATOR" -e PIN_NODE1="$PIN_NODE1" -e PIN_NODE2="$PIN_NODE2" -e ELASTICSEARCH="$ELASTICSEARCH" -e ELASTICSEARCH_PORT="$ELASTICSEARCH_PORT" -e ELASTICSEARCH_USER="$ELASTICSEARCH_USER" -e ELASTICSEARCH_PASSWORD="$ELASTICSEARCH_PASSWORD" -e BUILD_VERSION="$build_version" -e CI_STATUS="$status" -e CI_MINUTES_TIME="$ci_minutes_time" -e BENCHMARK_RUNNER_ID=$BENCHMARK_RUNNER_ID -e BENCHMARK_OPERATOR_ID="$BENCHMARK_OPERATOR_ID" -e BENCHMARK_WRAPPER_ID="$BENCHMARK_WRAPPER_ID" -e RUN_TYPE="perf_ci" -e TIMEOUT="3600" -e log_level="INFO" -v "$KUBECONFIG_PATH":"$CONTAINER_KUBECONFIG_PATH" --privileged "quay.io/ebattat/benchmark-runner:latest"
```
Comment (RobertKrawitz) [inline .github/workflows/Nightly_Perf_Env_CI.yml:299]: Same quoting issue as above.

File: benchmark_runner/common/oc/oc.py [hunk 5]
```diff
@@ -131,6 +131,13 @@ def is_odf_installed(self):
             return True
         return False
 
+    def get_odf_disk_count(self):
+        """
+        This method return odf disk count
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:136]: returns

File: benchmark_runner/common/oc/oc.py [hunk 6]
```diff
@@ -131,6 +131,13 @@ def is_odf_installed(self):
...
+        :return:
+        """
+        return int(self.run(f"{self.__cli} get pod -n openshift-storage | grep osd | grep -v prepare | wc -l"))
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:139]:
```
{self.__cli} get --no-headers pod ...
```

Comment (RobertKrawitz) [reply benchmark_runner/common/oc/oc.py:139]: Also, rather than
```
grep -v prepare | wc -l
```
use
```
grep -c -v prepare
```

File: tests/integration/benchmark_runner/common/oc/test_oc.py [hunk 7]
```diff
@@ -66,7 +66,17 @@ def test_oc_get_odf_version():
...
+
+
+def test_oc_get_odf_disk_count():
+    """
+    This method get odf disk count
+    :return:
+    """
+    oc = OC(kubeadmin_password=test_environment_variable['kubeadmin_password'])
+    oc.login()
+    assert oc.get_odf_disk_count() > 0
```
Comment (RobertKrawitz) [inline tests/integration/benchmark_runner/common/oc/test_oc.py:79]: Make sure that this works correctly if ODF is not installed.

File: tests/integration/benchmark_runner/common/oc/test_oc.py [hunk 8]
```diff
@@ -66,7 +66,17 @@ def test_oc_get_odf_version():
     """
     oc = OC(kubeadmin_password=test_environment_variable['kubeadmin_password'])
     oc.login()
-    oc.get_odf_version()
+    assert oc.get_odf_version()
+
+
+def test_oc_get_odf_disk_count():
+    """
+    This method get odf disk count
```
Comment (RobertKrawitz) [inline tests/integration/benchmark_runner/common/oc/test_oc.py:74]: gets

**Review by RobertKrawitz — COMMENTED** [review]
Nothing really big, but please fix these.

File: benchmark_runner/common/oc/oc.py (hunk 6 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py:139]: Fixed

File: benchmark_runner/common/oc/oc.py (hunk 5 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py:136]: Fixed

File: .github/workflows/Nightly_Perf_Env_CI.yml (hunk 4 above)
Comment (ebattat) [reply .github/workflows/Nightly_Perf_Env_CI.yml:299]: Fixed

File: .github/workflows/Nightly_Func_Env_CI.yml (hunk 2 above)
Comment (ebattat) [reply .github/workflows/Nightly_Func_Env_CI.yml:308]: We have the -e after, its the same

File: .github/workflows/Nightly_Func_Env_CI.yml (hunk 1 above)
Comment (ebattat) [reply .github/workflows/Nightly_Func_Env_CI.yml:286]: Fixed

File: tests/integration/benchmark_runner/common/oc/test_oc.py (hunk 8 above)
Comment (ebattat) [reply tests/integration/benchmark_runner/common/oc/test_oc.py:74]: Fixed

File: tests/integration/benchmark_runner/common/oc/test_oc.py (hunk 7 above)
Comment (ebattat) [reply tests/integration/benchmark_runner/common/oc/test_oc.py:79]: We must have ODF on our CI env

**Comment by ebattat** [comment]
@RobertKrawitz do you have more comments?

File: tests/integration/benchmark_runner/common/oc/test_oc.py (hunk 7 above)
Comment (RobertKrawitz) [reply tests/integration/benchmark_runner/common/oc/test_oc.py:79]: Our CI environment, yes, but this is a public repo, and someone may want to run without ODF.

**Comment by RobertKrawitz** [comment]
> @RobertKrawitz do you have more comments?

Only that if we want this to be general (beyond our CI environment) that the code should handle having no ODF present.  Perhaps an alternative would be a command line argument/environment variable to specify whether ODF is expected to be present.

File: tests/integration/benchmark_runner/common/oc/test_oc.py (hunk 7 above)
Comment (ebattat) [reply tests/integration/benchmark_runner/common/oc/test_oc.py:79]: Yes, you right. fixed

**Comment by ebattat** [comment]
@RobertKrawitz, fixed it

File: .github/workflows/Weekly_Perf_Env_Installer_CI.yml [hunk 9]
```diff
@@ -64,7 +64,7 @@ jobs:
           END
       - name: ▶ OCP assisted installer
         env:
-          INSTALL_OCP_VERSION: "latest-4.12.0-rc"
+          INSTALL_OCP_VERSION: "latest-4.12"
```
Comment (RobertKrawitz) [inline .github/workflows/Weekly_Perf_Env_Installer_CI.yml:67]: This belongs in a separate PR.

File: .github/workflows/Weekly_Perf_Env_Installer_CI.yml [hunk 10]
```diff
@@ -87,7 +87,7 @@ jobs:
           echo '>>>>>>>>>>>>>>>>>>>>>>>>>> OCP end step: ${{ matrix.step }}   >>>>>>>>>>>>>>>>>>>>>>>>>>>>'
       - name: ▶ Rerun OCP assisted install after failure
         env:
-          INSTALL_OCP_VERSION: "latest-4.12.0-rc"
+          INSTALL_OCP_VERSION: "latest-4.12"
```
Comment (RobertKrawitz) [inline .github/workflows/Weekly_Perf_Env_Installer_CI.yml:90]: This belongs in a separate PR.

File: tests/integration/benchmark_runner/common/oc/test_oc.py (hunk 7 above)
Comment (RobertKrawitz) [reply tests/integration/benchmark_runner/common/oc/test_oc.py:79]: This will still fail, because get_odf_disk_count won't return anything.  If you use my suggested fix above, it will be something like:

```
odf_disk_count = oc.get_odf_disk_count()
assert(odf_disk_count is None or odf_disk_count > 0)
```

File: benchmark_runner/common/oc/oc.py [hunk 11]
```diff
@@ -131,6 +131,14 @@ def is_odf_installed(self):
             return True
         return False
 
+    def get_odf_disk_count(self):
+        """
+        This method returns odf disk count
+        :return:
+        """
+        if self.is_odf_installed():
+            return int(self.run(f"{self.__cli} get --no-headers pod -n openshift-storage | grep osd | grep -cv prepare"))
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:140]: Needs to return a distinct value (probably None) if ODF is not installed.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
1) Please put the OCP version bump in a separate PR.
2) The logic of the get_odf_disk_count fix is not correct.

File: .github/workflows/Weekly_Perf_Env_Installer_CI.yml (hunk 9 above)
Comment (RobertKrawitz) [reply .github/workflows/Weekly_Perf_Env_Installer_CI.yml:67]: Change is already in main.  This should not have been committed here, but it should not cause a problem.

**Review by RobertKrawitz — APPROVED** [review]
